clang/lib/Sema/SemaType.cpp

```diff
@@ -8612,7 +8612,11 @@ static void HandleLifetimeBoundAttr(TypeProcessingState &State,
     CurType = State.getAttributedType(
         createSimpleAttr<LifetimeBoundAttr>(State.getSema().Context, Attr),
         CurType, CurType);
+    return;
   }
+  State.getSema().Diag(Attr.getLoc(), diag::err_attribute_wrong_decl_type_str)
+      << Attr << Attr.isRegularKeywordAttribute()
+      << "parameters and implicit object parameters";
```

Comment:
lifetime_capture_by is a similar attribute, we should apply the same check for it (on Line 8629), but it can be addressed in a follow-up patch.

Reply:
Sure! Let me create a follow-up PR, since that'll be a work for a different (although very similar) attribute.